deployment/terraform/engine.go

```diff
@@ -85,3 +86,20 @@ func (t *Terraform) runCommand(dst io.Writer, args ...string) error {
 	}
 	return nil
 }
+
+func checkTerraformVersion() error {
+	cmd := exec.Command("terraform", "version")
+	out, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("could not run terraform version command: %w", err)
+	}
+	re := regexp.MustCompile(`v\d+.?\d+`)
+	if !re.Match(out) {
+		return fmt.Errorf("could not parse terraform command output: %s", out)
+	}
+	version := string(re.FindAll(out, 1)[0])
+	if version != "v0.12" {
+		return fmt.Errorf("terraform v0.12 is required, you have %s", version)
+	}
```

Comment:
Technically this can become:

```go
	if re := regexp.MustCompile(`v\d+.?\d+`); !re.Match(out) {
		return fmt.Errorf("could not parse terraform command output: %s", out)
	} else if version := string(re.FindAll(out, 1)[0]); version != "v0.12" {
		return fmt.Errorf("terraform v0.12 is required, you have %s", version)
	}
```

But I won't enforce it since it might actually complicate the readability of the logic a bit. It's not always that easy to keep consistency :P

Reply:
```suggestion
	if re := regexp.MustCompile(`v\d+.?\d+`); !re.Match(out) {
		return fmt.Errorf("could not parse terraform command output: %s", out)
	}
	if version := string(re.FindAll(out, 1)[0]); version != "v0.12" {
		return fmt.Errorf("terraform v0.12 is required, you have %s", version)
	}
```

Reply:
I liked the approach 🥂